### PR TITLE
Fixes #14522 - add support for upstream openscap repo

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: manifests/plugin/openscap.pp [hunk 1]
```diff
@@ -7,6 +7,9 @@
 # $version::                    plugin package version, it's passed to ensure parameter of package resource
 #                               can be set to specific version number, 'latest', 'present' etc.
 #
+# $configure_openscap_repo::    Enable custom yum repo with packages needed for foreman_openscap,
```
Comment (domcleal) [inline manifests/plugin/openscap.pp]: smart_proxy_openscap?

File: manifests/plugin/openscap.pp [hunk 2]
```diff
@@ -44,9 +49,32 @@
...
+        $repo = $::operatingsystem ? {
+          'Fedora' => 'fedora',
+          default  => 'epel',
+        }
+
+        yumrepo { 'isimluk-openscap':
+          enabled  => 1,
+          gpgcheck => 0,
+          baseurl  => "http://copr-be.cloud.fedoraproject.org/results/isimluk/OpenSCAP/${repo}-${$::foreman_proxy::plugin::openscap::params::major_version}-\$basearch/",
+          before   => [ Foreman::Plugin['openscap'] ],
```
Comment (domcleal) [inline manifests/plugin/openscap.pp]: Foreman_proxy

File: manifests/plugin/openscap.pp [hunk 3]
```diff
@@ -44,9 +49,32 @@
...
+      default: {
+        fail("Unsupported osfamily ${::osfamily}")
+      }
+    }
+  }
+
   foreman_proxy::plugin { 'openscap':
     version => $version,
   } ->
+  foreman_proxy::plugin { 'openscap': } ->
```
Comment (domcleal) [inline manifests/plugin/openscap.pp]: again?

File: spec/classes/foreman_proxy__plugin__openscap_spec.rb [hunk 4]
```diff
@@ -33,6 +33,60 @@
...
+      context 'with enabled repository installation' do
+        let :params do
+          {
+            :configure_openscap_repo => true
+          }
+        end
+
+        it 'should fail' do
+          expect {
+            should contain_foreman__plugin('openscap')
```
Comment (domcleal) [inline spec/classes/foreman_proxy__plugin__openscap_spec.rb]: foreman_proxy

File: manifests/plugin/openscap.pp (hunk 3 above)
Comment (ares) [inline manifests/plugin/openscap.pp]: I messed up rebase, sorry for all such stupid mistakes

**Comment by ares** [comment]
it seems facts [can no longer](https://travis-ci.org/theforeman/puppet-foreman_proxy/jobs/121402718#L295) be used in `context` scope, any idea how to test this better?

**Comment by domcleal** [comment]
> it seems facts can no longer be used in context scope, any idea how to test this better?

facts is defined in a let block, and it only tests EL6, so remove the conditional.

**Comment by ares** [comment]
thanks @domcleal, tests are green now

**Comment by domcleal** [comment]
Merged, thanks @ares.

**Pull request merged**